frontend/src/components/Login.js

```diff
@@ -0,0 +1,136 @@
+
+  const dispatch = useDispatch()
+
+  // Form validation
+  const minimumNameLength = { minLength: 2, maxLength: 20 }
+  const minimumPasswordLength = { minLength: 5 }
+  const validEmail = { pattern: "^[a-z0-9._%+-]+@[a-z0-9.-]+.[a-z]{2,}$" }
+  //cannot start with @ and has to include the pattern of xxxx@xxx.se
+
+  // Toggle password visibility
+  const handleClickShowPassword = () => setShowPassword(!showPassword)
+
+  const handleLogin = (event) => {
+    event.preventDefault()
+    dispatch(login(name, email, password))
+    setName('')
+    setEmail('')
+    setPassword('')
+  }
+
+  const handleSignUp = (event) => {
+    event.preventDefault()
+    dispatch(signUp(name, email, password))
+    setName('')
+    setEmail('')
+    setPassword('')
+  }
+
+  return (
+    <Container>
+      <Wrapper>
+        {!accessToken ? (
+          <Form onSubmit={(event) => event.preventDefault()}>
+            <Title>Log in/Sign up</Title>
+            <SubTitle>to reveal your secret message...</SubTitle>
+            <TextField
+              required id="standard-required"  //adds *
+              label="Name"
+              inputProps={minimumNameLength}
+              value={name}
+              onChange={(event) => setName(event.target.value)}
+              helperText={name === "" ? 'min. 2 char' : ' '}
+            />
+            <div>
+              <InputLabel htmlFor="standard-adornment-password">Password*</InputLabel>
+              <Input
+                required={true}
+                id="standard-adornment-password"
+                type={showPassword ? "text" : "password"} //changed when icon is clicked
+                value={password}
+                fullWidth={true}
+                onChange={(event) => setPassword(event.target.value)}
+                inputProps={minimumPasswordLength}
+                endAdornment={
+                  <InputAdornment position="end">
+                    <IconButton
+                      aria-label="toggle password visibility"
+                      onClick={handleClickShowPassword}
+                    >
+                      {/* toggles */}
+                      {showPassword ? <Visibility /> : <VisibilityOff />}
+                    </IconButton>
+                  </InputAdornment>
+                }
+              />
+            </div>
+
+            <TextField
+              required id="standard-default"  //adds *
+              label="Email"
+              value={email}
+              onChange={(event) => setEmail(event.target.value)}
+              inputProps={validEmail}
+              helperText={email === "" ? 'x@xxx.xx' : ' '}
+            />
+
+            <ButtonWrapper>
+              <CustomButton
+                variant="contained"
+                color="default"
+                size="large"
+                type='submit'
+                onClick={handleSignUp}
+                text='Sign up'
+                disabled={!name || !password || !email}
```

Comment:
To be extra clear to the user you could also add number of characters needed in password here on the validation.